source/jormungandr/jormungandr/scenarios/experimental.py

```diff
@@ -146,13 +146,26 @@ def call_kraken(self, request_type, request, instance, krakens_call):
         for dep_mode, arr_mode in krakens_call:
             #todo: this is probably shared between multiple thread
             self.nb_kraken_calls += 1
+            direct_path = self._get_direct_path(instance,
+                                                dep_mode,
+                                                request['origin'],
+                                                request['destination'],
+                                                request['datetime'],
+                                                request['clockwise'])
+            if direct_path.journeys:
+                journey_parameters.direct_path_duration = direct_path.journeys[0].durations.total
```

Comment:
Since the journey_parameters is keep between call to kraken you need to reset the value of `direct_path_duration` if no direct path has been found